src/hotspot/cpu/riscv/riscv_v.ad

```diff
@@ -2077,3 +2076,20 @@ instruct vclearArray_reg_reg(iRegL_R29 cnt, iRegP_R28 base, Universe dummy,
 
   ins_pipe(pipe_class_memory);
 %}
+
+// Vector Load Const
+instruct vloadcon(vReg dst, immI0 src) %{
+  match(Set dst (VectorLoadConst src));
+  ins_cost(VEC_COST);
+  format %{ "vloadcon $dst\t# generate iota indices" %}
+  ins_encode %{
+    BasicType bt = Matcher::vector_element_basic_type(this);
+    Assembler::SEW sew = Assembler::elemtype_to_sew(bt);
+    __ vsetvli(t0, x0, sew);
+    __ vid_v(as_VectorRegister($dst$$reg));
+    if (is_floating_point_type(bt)) {
+      __ vfcvt_f_x_v(as_VectorRegister($dst$$reg), as_VectorRegister($dst$$reg));
```

Comment:
You might want to distinugish between float and double for 'bt' here. Since vfcvt.f.x.v only convert signed integer to float.

Reply:
Hi @RealFYang , thanks for the review! Since `vid.v` generates a sequence of integers, if `bt` is a floating point type, it should be converted and stored in each element of the vector.

Reply:
Hi, I double checked the RVV spec and I think you are right here. Change looks good then.